code/datums/components/item_two_handed.dm

```diff
@@ -0,0 +1,339 @@
+	if(offhand_item && istype(offhand_item))
+		offhand_item.unwield()
+
+/**
+ * on_attack triggers on attack with the parent item
+ */
+/datum/component/two_handed/proc/on_attack(obj/item/source, mob/living/target, mob/living/user)
+	if(wielded && attacksound)
+		var/obj/item/parent_item = parent
+		playsound(parent_item.loc, attacksound, 50, TRUE)
+
+/**
+ * on_update_icon triggers on call to update parent items icon
+ *
+ * Checked in the order listed, first found is used
+ * Invokes the Icon Update Callback with wield status
+ * Updates the icon using icon_prefix if set
+ */
+/datum/component/two_handed/proc/on_update_icon(datum/source)
+	if(icon_update_callback)
+		icon_update_callback.Invoke(wielded)
+	else if(icon_prefix)
+		var/obj/item/parent_item = parent
+		if(parent_item)
+			parent_item.icon_state = "[icon_prefix][wielded]"
+
+/**
+ * try_wield tries to wield the item
+ */
+/datum/component/two_handed/proc/try_wield(datum/source, mob/user)
+	wield(user)
+
+/**
+ * try_unwield attempts to unwield the item
+ */
+/datum/component/two_handed/proc/try_unwield(datum/source, mob/user, show_message=TRUE)
+	unwield(user, show_message)
+
+/**
+ * on_swap_hands Triggers on swapping hands, blocks swap if the other hand is busy
+ */
+/datum/component/two_handed/proc/on_swap_hands(mob/user, obj/item/held_item)
+	if(!held_item)
+		return
+	var/datum/component/two_handed/comp_twohand = held_item.GetComponent(/datum/component/two_handed)
+	if(comp_twohand && comp_twohand.wielded)
+		return COMPONENT_BLOCK_SWAP
+
+/**
+ * The offhand dummy item for two handed items
+ *
+ */
+/obj/item/offhand
+	name = "offhand"
+	icon_state = "offhand"
+	w_class = WEIGHT_CLASS_HUGE
+	item_flags = ABSTRACT
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+	var/wielded = FALSE // Off Hand tracking of wielded status
+
+/obj/item/offhand/Destroy()
+	wielded = FALSE
+	return ..()
+
+// Only utilized by dismemberment since you can't normally switch to the offhand to drop it.
+/obj/item/offhand/dropped(mob/living/user, show_message=TRUE)
+	SHOULD_CALL_PARENT(0)
+
+	// Call the held object
+	var/obj/item = user.get_active_held_item()
+	if(item)
+		var/datum/component/two_handed/comp_twohand = item.GetComponent(/datum/component/two_handed)
+		if(comp_twohand)
+			SEND_SIGNAL(item, COMSIG_TRY_TWOHANDED_UNWIELD, user, show_message)
+			// Drop item if off hand is dropped and the item requies both hands
+			if(comp_twohand.require_twohands)
+				user.dropItemToGround(item)
+
+	// delete on drop
+	if(!QDELETED(src))
+		qdel(src)
+
+/obj/item/offhand/equipped(mob/user, slot)
+	..()
+	if(wielded && !user.is_holding(src))
+		unwield(user)
+
+/obj/item/offhand/proc/unwield()
+	if(wielded)
+		wielded = FALSE
+		qdel(src)
+
+// You should never be able to do this in standard use of two handed items. This is a backup for lingering offhands.
+/obj/item/offhand/attack_self(mob/living/carbon/user)
```

Comment:
You shouldn't need this proc, if it doesn't properly get dealt with by the component let it sit there so that there's indication something went wrong

Reply:
You are right I have removed this on the next update.